homeassistant/components/hdmi_cec/media_player.py

```diff
@@ -52,13 +55,18 @@
 ENTITY_ID_FORMAT = DOMAIN + ".{}"
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+def setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
     """Find and return HDMI devices as +switches."""
-    if ATTR_NEW in discovery_info:
+    if discovery_info and ATTR_NEW in discovery_info:
         _LOGGER.debug("Setting up HDMI devices %s", discovery_info[ATTR_NEW])
         entities = []
         for device in discovery_info[ATTR_NEW]:
-            hdmi_device = hass.data.get(device)
+            hdmi_device = hass.data[device]
```

Comment:
The next line expects the data to be available, so it can query the corresponding `logical_address`

Reply:
Side note: I noticed that the device is registered at the top level of `hass.data` in https://github.com/home-assistant/core/blob/4f3fa0dbbe63677be4bccf86f9db872e344afcd4/homeassistant/components/hdmi_cec/__init__.py#L335-L336. Should this be adjusted to `hass.data[DOMAIN][device]` in a follow-up PR?